Created WHSSDaltas_v1 and v4.01 and updated v1.01 -> v4

matsalan opened the pull request:
"I have worked with the WHSSDatlas and created brainAtlasVersion, and parcellationEntityVersion for the v1 and v4.01, and parcellationEntity for new v4.01 entries.
The parcellationEntity has been updated compared to the other to include hasParent, and abbreviation.
The parcellationEntityVersion have been created in the same format as the other based on the v1 and v4.01 ilf files. However I have also added more information. AnchorPoint have been filled with details from nii files extracted with a program written by Gergely Csucs. hasParent have been filled in as I managed to extract the hierarchy from ilf files with the program. I also updated a versionInnovation document with information from the Waxholm atlas and used it to fill in the criteria and versionInnovation(either unchanged, revised or new). The criteria also had to be updated to a longer reference which is also done.
The brainAtlasVersion has no changes to the template and is filled in with updated information compared to the other brainAtlasVersions".
I also updated the otherversions to include anchorpoint, criteria, version innovation and for v3 -> v4 included the abbreviation.

matsalan committed:
Add the WHS v4.01 to the github

Includes upto date information, in addition to criteria and anchorpoint information.

matsalan committed:
Merge branch 'v3' of https://github.com/matsalan/openMINDS_SANDS into v3

matsalan committed:
Revert "Add the WHS v4.01 to the github"

This reverts commit 0a5050738ba07b3b67a57202c2ab1e3146c855d1.

matsalan committed:
Added versions for WHSDatlas_v4.01

I  made a python program that wrote the parcellationEntitiyVersions. however i also made the program include criteria, anchorpoint and versioninnovation. Version innovation is not quite finished until revisions are made for the revision version innovation comments.

matsalan committed:
Created brainAtlasVerson for WHSSDatlas v4.01 and 1

I wrote a program to create brain atlas version for version 4.01 and 1. v1 varies from v1.01 with version innovation as it is the first, and changed howTocite. v4.01 had mostly the same only with minor changes compared to v4.01

matsalan committed:
Wrote parcellationEntityVersion for v1 and changed a writing mistake for v4.01

I created a parcellationEntityVersion versions for the v1 of WHSDatlas with both criteria, anchorpoint, alternateName based from v.101, hasParent and versionInnovation,

I also modified the v4.01 parcellation version to remove ]from the end of the sources due to a coding mistake

matsalan committed:
Made parcellationEntity for new v4.01 regions.

matsalan committed:
Changed names to lower letter

Apperantly some of the names in the ilf file were capitalized so i had to change them to lower case

matsalan committed:
Update WHSSDatlas_anteriorOlfactoryAreaUnspecified.jsonld

matsalan committed:
Updated the parcellationEntity to include new versions.

Made a program that includes the addition of v1 and v4.01

matsalan committed:
Updated v1.01 -> v4 to include anchorpoint, criteria and version innovation

I made another program that appends criteria based on the waxholm website to include version innovation changes either revised unchanged or first deliniated in the version. It also includes the criteria and the anchorpoint, as well as appends the abbreviation for v3 -> v4

matsalan committed:
Updated the documents to have \n at the last line

matsalan committed:
Added newline to parcellationEntity

added \n at the end of the files

lzehl commented on instances/atlas/brainAtlasVersion/WHSSDatlas_v1.jsonld:
```diff
@@ -0,0 +1,283 @@
+    "copyright": null,
+    "custodian": null,
+    "description": null,
+    "digitalIdentifier": {
```
this should be null since this is the RRID of the atlas as concept

lzehl commented on instances/atlas/brainAtlasVersion/WHSSDatlas_v1.jsonld:
```diff
@@ -0,0 +1,283 @@
+        "ontologyIdentifier": null
+    },
+    "homepage": "https://www.nitrc.org/projects/whs-sd-atlas/",
+    "howToCite": "Please refer to the atlas by its RRID: SCR_017124, and cite the following publications:\n- Papp EA, Leergaard TB, Calabrese E, Johnson GA, Bjaalie JG (2014) Waxholm Space atlas of the Sprague Dawley rat brain. NeuroImage 97, 374-386. DOI: https://doi.org/10.1016/j.neuroimage.2014.04.001",
```
sorry for going back and forth with this. Considering documentation displays let's try to keep the text simple:

"Please refer to the atlas by its [RRID:SCR_017124](https://scicrunch.org/resolver/SCR_017124), and cite the following publications: [Papp et al. (2014)](https://doi.org/10.1016/j.neuroimage.2014.04.001)"

lzehl commented on instances/atlas/brainAtlasVersion/WHSSDatlas_v4.01.jsonld:
```diff
@@ -0,0 +1,740 @@
+        "ontologyIdentifier": null
+    },
+    "homepage": "https://www.nitrc.org/projects/whs-sd-atlas/",
+    "howToCite": "Please refer to the atlas by its RRID: SCR_017124, and cite the following publications:\n- Papp EA, Leergaard TB, Calabrese E, Johnson GA, Bjaalie JG (2014) Waxholm Space atlas of the Sprague Dawley rat brain. NeuroImage 97, 374-386. DOI: https://doi.org/10.1016/j.neuroimage.2014.04.001\n- Heidi Kleven, Ingvild Bjerke, Francisco Clasc\u00c3\u00a1 et al. Waxholm Space atlas of th ...
```
cf. comment in v1

lzehl commented on instances/atlas/parcellationEntityVersion/WHSSDatlas_v4/WHSSDatlas_v4_superiorPeriolivaryRegion.jsonld:
```diff
@@ -1,58 +1,71 @@
+            "criteriaType": {
+                "@id": "https://openminds.ebrains.eu/instances/annotationCriteriaType/deterministicAnnotation"
+            },
+            "inspiredBy": null,
```
```json
"inspiredBy": [
  {
    "@id": "URLTOFILE"
  }
]
```

matsalan replied:
What file is it inspired by, which URL should i use? @lzehl

lzehl replied:
the URL to the actual file where it can be downloaded. Example:

"https://www.nitrc.org/frs/download.php/13400/MBAT_WHS_SD_rat_atlas_v4.01.zip"

matsalan replied:
@lzehl This should be resolved now, i linked to the "https://www.nitrc.org/frs/?group_id=1081"

lzehl reviewed (changes requested):
+ double check if "hasParent" link is present for PEs (!) and PEVs
+ double check if parent instances are registered as PEs
+ please double check which PEVs should exist per PE (v1 - v4.01?)

criteria registration needs to be removed for now. registration options for criteria need to be reevaluated.

matsalan committed:
Update WHSSDatlas_v1.jsonld

Set digitalIdentifier to null, and changed the reference to a shorter version.

matsalan committed:
Update WHSSDatlas_v4.01.jsonld

Changed refrence format

matsalan committed:
Added hyperlink? WHSSDatlas_v1.jsonld

matsalan committed:
Additional Hyperlink WHSSDatlas_v1.jsonld

lzehl commented on instances/atlas/brainAtlasVersion/WHSSDatlas_v1.jsonld:
```diff
@@ -254,7 +254,7 @@
     },
     "homepage": "https://www.nitrc.org/projects/whs-sd-atlas/",
-    "howToCite": "Please refer to the atlas by its RRID:SCR_017124, and cite the following publications: <a href='https://doi.org/10.1016/j.neuroimage.2014.04.001'>Papp et al (2014)</a>",
+    "howToCite": "Please refer to the atlas by its <a href='https://scicrunch.org/resolver/SCR_017124'>RRID:SCR_017124</a>, and cite the following publications: <a href='https://doi.org/10.1016/j.neuroimage.2014.04.001'>Papp et al (2014)</a>",
```
please use as I posted in my comments markdown syntax.

"Please refer to the atlas by its [RID:SCR_017124](https://scicrunch.org/resolver/SCR_017124), and cite the following publications: [Papp et al (2014)](https://doi.org/10.1016/j.neuroimage.2014.04.001)"

matsalan replied:
> 
I do not know how to write in markdown syntax, can you post exactly how you want me to write it in the document. I tied with <a href = ''> but i cannot use "" as it stops the quote. I have tried fo copy paste what you wrote, but the hyperlink is not conserved. Also do you mean to close the [.

lzehl replied:
`"Please refer to the atlas by its [RRID:SCR_017124](https://scicrunch.org/resolver/SCR_017124), and cite the following publications: [Papp et al (2014)](https://doi.org/10.1016/j.neuroimage.2014.04.001)"`

lzehl replied:
I already did (with one type for the missing ])... here again

lzehl replied:
```json
"howToCite": "Please refer to the atlas by its [RRID:SCR_017124](https://scicrunch.org/resolver/SCR_017124), and cite the following publications: [Papp et al (2014)](https://doi.org/10.1016/j.neuroimage.2014.04.001)"
```

matsalan replied:
> I already did... here again

Sorry, the full format showed up now, i am not quite sure why it was displayed different earlier. Anyway, i pushed the change at v1 brain atlas version, is it correct now?

lzehl replied:
It is except for the error that you copied over from my first example. It's called RRID not RID

matsalan replied:
Are v1 and v4.01 brain atlas version how to cite correct now?

matsalan committed:
Added hyperlinks WHSSDatlas_v4.01.jsonld

matsalan commented:
> 
I talked with heidi regarding which PEVs should exist in PE, the 4th ventricle is not deliniated in v 1->2 as it is called the ventricular system. Unless you see a version one and not v1.01 and v2 then the pE is not suppose to exist prior to v3.

matsalan committed:
removed criteria and added inspiredBy in  parcellationEntityVersion

Removed the criteria and added link to the waxholm nitiric atlas site

matsalan committed:
Changed to replace "-" with "'"

This allows the name to match the other versions with small capital letter

matsalan committed:
Updated pE to include hasParent

Based on the hasVersion names, the program added the Parent from the ilf documentation of the corresponding version. This was the hasParent is based on the newest avaliable ilf document for the version it is registered.

matsalan committed:
Hyperlink update WHSSDatlas_v1.jsonld

matsalan committed:
Hyperlink again WHSSDatlas_v1.jsonld

matsalan committed:
fixed RRID WHSSDatlas_v1.jsonld

lzehl commented:
> > 
> 
> I talked with heidi regarding which PEVs should exist in PE, the 4th ventricle is not deliniated in v 1->2 as it is called the ventricular system. Unless you see a version one and not v1.01 and v2 then the pE is not suppose to exist prior to v3.

This is a wrong assumption. The PEV may not exist prior to v3, but the structure definitely has to exist as PE (which is a version independent representation)

matsalan committed:
Fixed hyperlink WHSSDatlas_v4.01.jsonld

matsalan commented:
> > > 
> > 
> > 
> > I talked with heidi regarding which PEVs should exist in PE, the 4th ventricle is not deliniated in v 1->2 as it is called the ventricular system. Unless you see a version one and not v1.01 and v2 then the pE is not suppose to exist prior to v3.
> 
> This is a wrong assumption. The PEV may not exist prior to v3, but the structure definitely has to exist as PE (which is a version independent representation)

Okay, however i based these updates on the ilf documents and the 4th ventricle is not deliniated in the v1, v1.01 nor v2 ilf file then it will not get updated from the program atleast unless i can get some other documentation that shows it is deliniated in these versions. I do not know where to find information of the atlas versions that is not in the ilf documentation.

matsalan commented:
> > > > 
> > > 
> > > 
> > > I talked with heidi regarding which PEVs should exist in PE, the 4th ventricle is not deliniated in v 1->2 as it is called the ventricular system. Unless you see a version one and not v1.01 and v2 then the pE is not suppose to exist prior to v3.
> > 
> > 
> > This is a wrong assumption. The PEV may not exist prior to v3, but the structure definitely has to exist as PE (which is a version independent representation)
> 
> Okay, however i based these updates on the ilf documents and the 4th ventricle is not deliniated in the v1, v1.01 nor v2 ilf file then it will not get updated from the program atleast unless i can get some other documentation that shows it is deliniated in these versions. I do not know where to find information of the atlas versions that is not in the ilf documentation.

I think we agree, but the wording may have been wrong. I agree that 4th ventricle PE should exist, but what is meant is that the hasVersion list for 4th ventrilce only includes v3-> v4.01, as it was first deliniated in v3 and the 4th ventricle hasVersion list  should not include v1 -> v2 as it was not deliniated in those versions.

matsalan committed:
Corrected indentation of the WHSSDaltas

matsalan committed:
Revert "Corrected indentation of the WHSSDaltas"

This reverts commit 1de11df64a5ea1446d6caefc41501051d7fec3d9.

lzehl commented:
> > > > > 
> > > > 
> > > > 
> > > > I talked with heidi regarding which PEVs should exist in PE, the 4th ventricle is not deliniated in v 1->2 as it is called the ventricular system. Unless you see a version one and not v1.01 and v2 then the pE is not suppose to exist prior to v3.
> > > 
> > > 
> > > This is a wrong assumption. The PEV may not exist prior to v3, but the structure definitely has to exist as PE (which is a version independent representation)
> > 
> > 
> > Okay, however i based these updates on the ilf documents and the 4th ventricle is not deliniated in the v1, v1.01 nor v2 ilf file then it will not get updated from the program atleast unless i can get some other documentation that shows it is deliniated in these versions. I do not know where to find information of the atlas versions that is not in the ilf documentation.
> 
> I think we agree, but the wording may have been wrong. I agree that 4th ventricle PE should exist, but what is meant is that the hasVersion list for 4th ventrilce only includes v3-> v4.01, as it was first deliniated in v3 and the 4th ventricle hasVersion list should not include v1 -> v2 as it was not deliniated in those versions.

@matsalan I understand. this is fine. generally I couldn't find any parent relations between the PEVs and the PEs though last time I checked (maybe I looked in the wrong commit though). Just double check that all PEVs have a PE as parent structure.

matsalan committed:
Corrected indendation to 2

matsalan committed:
Added newline to v1 pEV

matsalan committed:
Updated inspiredBy link, fixed names

Changed it from direct download to the download directory so those interested can choose what type of file they will download.
Also some names still had symbols in them so i changed it to remove those symbols especially () with reticular (Pre) thalamic

matsalan committed:
Update parents of other versions

Went through all the hasParent to check if they are correct and these are the changes that should be done according to the ilf files

UlrikeS91 commented:
I had a look at the content of this PR and found quite a few errors. I believe this is a combination of both new and existing errors in the files. Since we need the Waxholm instances moved to the new openMINDS_instances repo as soon as possible, I discussed with @aeidi89 to close this PR and migrate the instances as they are.  

I will use the content of this PR (and huge thanks to @matsalan for this great work!) to update the instances following the migration some time in the near future instead.
